katsdpingest/katsdpingest/test/test_sigproc.py

```diff
@@ -225,17 +241,16 @@ def _test_big(self, context, queue, excise):
 
         # Perform the operation on the host
         kept_vis = vis_in[:, channel_range.start : channel_range.stop]
-        kept_flags = flags_in[:, channel_range.start : channel_range.stop] \
-            | channel_flags[np.newaxis, channel_range.start : channel_range.stop] \
-            | baseline_flags[:, np.newaxis]
+        kept_flags = flags_in[:, channel_range.start : channel_range.stop]
         if excise:
             flagged_weights = weights_in * ((kept_flags == 0) + FLAG_SCALE)
             # unflagged inputs need the UNFLAGGED_BIT set
-            kept_flags |= np.where(kept_flags, 0, UNFLAGGED_BIT).astype(np.uint8)
+            kept_flags = kept_flags | np.where(kept_flags, 0, UNFLAGGED_BIT).astype(np.uint8)
```

Comment:
Why not in-place?

Reply:
kept_flags starts as a view of flags_in, and I didn't want to modify flags_in in case that causes problems for the caller.